external-table/src/pxfuriparser.c

```diff
@@ -291,10 +290,13 @@ GPHDUri_verify_no_duplicate_options(GPHDUri *uri)
 	foreach(option, uri->options)
 	{
 		OptionData *data = (OptionData *) lfirst(option);
-#if PG_VERSION_NUM >= 90400
-		Value	   *key = makeString(asc_toupper(data->key, strlen(data->key)));
+
+// For GP 5.x latest ( 5.29.8) the server_version_num is 80323.
+// For 6.x or later str_toupper doesn't work as it has different signature ( one extra parameter) so the asc_toupper will work for 6.x or 7.x
```

Comment:
why is asc_toupper an acceptable replacement. are there caveats that we may need to be aware of? 

nit: 
```suggestion
// For 6.x or later str_toupper doesn't work as it has different signature (one extra parameter) so the asc_toupper will work for 6.x or 7.x

Reply:
I will look into it if we can still use `str_toupper` but to answer your question, the change was made to resolve any compilation issues.

Reply:
The function `asc_toupper` is an ASCII-only function and is available in GPDB 5. Since this is being used to verify the options provided in the PXF location URL, we might be able to say only ASCII characters may be used. But since we accept a server name which is used to find a directory in the file system this might be overly restrictive.

Since we've been using `asc_toupper` with GPDB6 all along, I don't think this is a change in behavior.